File: Applications/Book/book/03-locate-files/python/EpicStoreManager.py
import os, platform, json

def init():
    def getPath():
        system = platform.system()
        if system == 'Windows':
            # windows paths
            home = os.getenv('ALLUSERSPROFILE')
            search = ['Epic/EpicGamesLauncher']
            paths = [os.path.join(home, path, 'Data') for path in search]
        elif system == 'Linux':
            # linux paths
            home = os.path.expanduser('~')
            search = ['?GOG?']
            paths = [os.path.join(home, path, 'Data') for path in search]
        elif system == 'Darwin':
            # mac paths
            home = '/Users/Shared'
            search = ['Epic/EpicGamesLauncher']
            paths = [os.path.join(home, path, 'Data') for path in search]
        else: raise Exception(f'Unknown platform: {system}')
        return next(iter(x for x in paths if os.path.isdir(x)), None)
    
    # get dbPath
    root = getPath()
    if root == None: return
    dbPath = os.path.join(root, 'Manifests')
    # query games
    for s in [s for s in os.listdir(dbPath) if s.endswith('.item')]:
        with open(os.path.join(dbPath, s), 'r') as f:
            # add appPath if exists
            appPath = json.loads(f.read())['InstallLocation']
            if os.path.isdir(appPath): epicAppPaths[s[:-5]] = appPath

epicAppPaths = {}
init()
# print(epicAppPaths)

File: Context/GameSpec.Base/StoreManagers/EpicStoreManager.cs
using Microsoft.Win32;
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Runtime.InteropServices;
using System.Text.Json;

namespace GameSpec.StoreManagers
{
    /// <summary>
    /// EpicStoreManager
    /// </summary>
    internal static class EpicStoreManager
    {
        static Dictionary<string, string> AppPaths = new Dictionary<string, string>();

        public static bool TryGetPathByKey(string key, out string path) => AppPaths.TryGetValue(key, out path);

        static EpicStoreManager()
        {
            var root = GetPath();
            if (root == null) return;
            var dbPath = Path.Combine(root, "Manifests");
            foreach (var s in Directory.EnumerateFiles(dbPath).Where(s => s.EndsWith(".item")))
                AppPaths.Add(Path.GetFileNameWithoutExtension(s), JsonSerializer.Deserialize<JsonElement>(File.ReadAllText(s)).GetProperty("InstallLocation").GetString());
        }

        static string GetPath()
        {
            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
            {
                var home = Environment.GetFolderPath(Environment.SpecialFolder.CommonApplicationData);
                return new[] { @"Epic\EpicGamesLauncher" }
                    .Select(path => Path.Join(home, path, "Data"))
                    .FirstOrDefault(Directory.Exists);
            }
            else if (RuntimeInformation.OSDescription.StartsWith("android-")) return null;
            else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
            {
                var home = Environment.GetFolderPath(Environment.SpecialFolder.UserProfile);
                return new[] { "Epic/EpicGamesLauncher" }
                    .Select(path => Path.Join(home, path, "Data"))
                    .FirstOrDefault(Directory.Exists);
            }
            else if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
            {
                var home = "/Users/Shared";
                return new[] { "Epic/EpicGamesLauncher" }
                    .Select(path => Path.Join(home, path, "Data"))
                    .FirstOrDefault(Directory.Exists);
            }
            throw new PlatformNotSupportedException();
        }
    }
}

